### ISSUE TITLE
sprintf related compiler warnings

### ISSUE THREAD

**Issue by jorge-leon** [hughbarney/femto#13]
Compiling with gcc version 12.2.0 (Debian 12.2.0-14)  I get the following warnings, the second, third and fourth seem worth investigation. 

```
undo.c: In function ‘dump_undos’:
undo.c:470:71: warning: ‘%s’ directive writing up to 20 bytes into a region of size between 0 and 60 [-Wformat-overflow=]
  470 |                         sprintf(report_line, "%03d %6s %9ld %4d %s -> %s\n",  ++count, get_undo_type_name(prev), prev->u_point, size, buf1, buf2);
      |                                                                       ^~                                                                    ~~~~
undo.c:470:25: note: ‘sprintf’ output between 32 and 117 bytes into a destination of size 90
  470 |                         sprintf(report_line, "%03d %6s %9ld %4d %s -> %s\n",  ++count, get_undo_type_name(prev), prev->u_point, size, buf1, buf2);
      |                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

utils.c: In function ‘make_buffer_name_uniq’:
utils.c:45:34: warning: ‘%d’ directive writing between 1 and 10 bytes into a region of size between 1 and 17 [-Wformat-overflow=]
   45 |                 sprintf(bufn, "%s%d", basen, num++);
      |                                  ^~
utils.c:45:31: note: directive argument in the range [0, 2147483646]
   45 |                 sprintf(bufn, "%s%d", basen, num++);
      |                               ^~~~~~
utils.c:45:17: note: ‘sprintf’ output between 2 and 27 bytes into a destination of size 17
   45 |                 sprintf(bufn, "%s%d", basen, num++);
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

```

**Comment by hughbarney** [comment]
agree, should be fixed. These did not used to get reported.  I think the compiler just got better at detecting things.  Happy for a PR to be raised.

**Comment by jorge-leon** [comment]
Can be closed with #20

**Comment by hughbarney** [comment]
fixed in v2.15
